lib/world_cup.rb

```diff
@@ -0,0 +1,75 @@
+require './lib/player'
+require './lib/team'
+
+class WorldCup
+  attr_accessor :year,
+                :teams
+  def initialize(year, teams)
+    @year = year
+    @teams = teams
+  end
+
+  def active_players_by_position(position)
+    teams = @teams.select {|team| team.eliminated == false }
+    players = []
+    teams.each do |team|
+      players << team.players_by_position(position)
+    end
+    players.flatten
+  end
+
+  def all_players(active=false)
+    teams = @teams.select {|team| team.eliminated == false } if active == true
+    teams = @teams if active == false
+    defenders =  []
+    teams.each do |team|
+      defenders << team.players_by_position(:defender)
+    end
+    defenders.flatten!
+
+
+    defenders.map! {|player|  player.name}
+    defenders.sort!
+
+    forwards = []
+    teams.each do |team|
+      forwards << team.players_by_position(:forward)
+    end
+    forwards.flatten!
+    forwards.map! {|player|  player.name}
+    forwards.sort!
+
+    midfielders = []
+    teams.each do |team|
+      midfielders << team.players_by_position(:midfielder)
+    end
+    midfielders.flatten!
+    midfielders.map! {|player|  player.name}
+    midfielders.sort!
```

Comment:
Seems like a lot of duplication in lines 24 - 48. Breaking it out into a method would allow you to provide an additional layer of abstraction, name this thing, and then make changes in a single place if you decide you'd like to change something about how this works in the future.

One other thing to consider: what happens if we add a position? As this works currently, we need to add another seven lines. If we extract this to another method there's a decent chance that we don't have to add any.